Rover/sailboat.h

```diff
@@ -18,6 +18,15 @@
 */
 class Sailboat
 {
+
+   friend class Rover;
+   friend class Mode;
+   friend class ModeManual;
+   friend class ModeHold;
+   friend class ModeLoiter;
+   friend class ModeAcro;
+   friend class RC_Channel_Rover;
```

Comment:
I don't think these are needed for this PR? There are no changes to most of those classes.

Reply:
I modified Sailboat class member function access to make all the non const Sailboat member functions private and so those classes calling private Sailboat functions ( ie Sailboat state modifying functions)  require friend access to Sailboat. The overall idea is limit who can change the state as much as possible. If functions are public, anyone can change state thus making debugging harder.  The precedent is similar use of friends in Rover class https://github.com/ArduPilot/ardupilot/blob/master/Rover/Rover.h#L77 

Shall I revert it to those non const member functions of Sailboat being public?

Reply:
Ah, I see. I had not appreciated the change in scope of those functions. I think my preference would be to keep the current structure. 

They key difference is that all of the params and internal state variables are currently private. So cannot be changed externally. It we add friends then they can all reach in any change things they probably shouldn't. EG you should call `set_motor_state` not just change `motor_state`. That is currently enforced.

If there are things that should defiantly not be able to change anything (those class which you did not have to add as friends) then we could go via a const reference to the class. But i'm not really sure there are any that would need this?